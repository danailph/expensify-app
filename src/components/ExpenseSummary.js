import React from 'react'
import {connect} from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'


export const ExpenseSummary = ({count, formatedTotal}) => (
  <div>
    <h1>Viewing {count} {count === 1 ? 'expense':'expenses' } totalling {formatedTotal}</h1>
  </div>
)

const mapStateToProps = (state) => {
  const filteredExpenses  = selectExpenses(state.expenses, state.filters)
  const rawTotal = selectExpensesTotal(filteredExpenses)
  return {
    count: filteredExpenses.length,
    formatedTotal: numeral(rawTotal /  100).format('$0,0.00')
  }
}

export default connect(mapStateToProps)(ExpenseSummary);
