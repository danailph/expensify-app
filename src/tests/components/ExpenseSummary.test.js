import React from 'react'
import { shallow } from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'


test('should render ExpenseSummary correctly with one expense',() => {
  const wrapper = shallow(<ExpenseSummary count={1} formatedTotal={1000}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly with multiple expenses',() => {
  const wrapper = shallow(<ExpenseSummary count={3} formatedTotal={3000}/>)
  expect(wrapper).toMatchSnapshot()
})
