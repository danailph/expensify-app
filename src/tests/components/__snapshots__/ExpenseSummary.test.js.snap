// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    3000
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    1000
  </h1>
</div>
`;
